src/field-accessor.ts

```diff
@@ -267,6 +270,27 @@ export class FieldAccessor<R, V> extends AccessorBase implements IAccessor {
     return this.errorValue === undefined;
   }
 
+  @computed
+  get isDirty(): boolean {
+    if (this.addMode) {
+      return false;
+    }
+    if (Array.isArray(this.value)) {
+      return toJS(this.value).length !== this._originalValue.length;
```

Comment:
Is this a proper dirty check? Because it looks like it only checks if the array is equal in length. So if I change one value in the array but keep it the same length, will it correctly flag as dirty?

Reply:
Good catch! I came across this issue while implementing the restore functionality.
This has been fixed in that PR by comparing the values in its stringified form when the lengths are the same.